lib/firmata.js

```diff
@@ -1690,6 +1719,337 @@ Board.prototype.pingRead = function(opts, callback) {
+    }
+    if (opts.invertPins.indexOf(opts.motorPin4) !== -1) {
+      pinsToInvert |= 0x08;
+    }
+    if (opts.invertPins.indexOf(opts.enablePin) !== -1) {
+      pinsToInvert |= 0x10;
+    }
+  }
+
+  data.push(pinsToInvert);
+  data.push(END_SYSEX);
+  this.transport.write(new Buffer(data));
+
+};
+
+/**
+ * Asks the arduino to set the stepper position to 0
+ * Note: This is not a move. We are setting the current position equal to zero
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ */
+
+Board.prototype.accelStepperZero = function(deviceNum) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x01, // STEPPER_ZERO from firmware
+    deviceNum,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Asks the arduino to move a stepper a number of steps
+ * (and optionally with and acceleration and deceleration)
+ * speed is in units of steps/sec
+ * @param {number} deviceNum Device number for the stepper (range 0-5)
+ * @param {number} steps Number of steps to make
+ */
+Board.prototype.accelStepperStep = function(deviceNum, steps, callback) {
+  
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x02, // STEPPER_STEP from firmware
+    deviceNum
+  ];
+
+  Array.prototype.push.apply(data, Board.encode32BitSignedInteger(steps));
+  data.push(END_SYSEX);
+
+  this.transport.write(new Buffer(data));
+  
+  if (callback) {
+    this.once("stepper-done-" + deviceNum, callback);
+  }
+
+};
+
+/**
+ * Asks the arduino to move a stepper to a specific location
+ * @param {number} deviceNum Device number for the stepper (range 0-5)
+ * @param {number} position Desired position
+ */
+Board.prototype.accelStepperTo = function(deviceNum, position, callback) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x03, // STEPPER_TO from firmware
+    deviceNum
+  ];
+
+  Array.prototype.push.apply(data, Board.encode32BitSignedInteger(position));
+  data.push(END_SYSEX);
+
+  this.transport.write(new Buffer(data));
+
+  if (callback) {
+    this.once("stepper-done-" + deviceNum, callback);
+  }
+
+};
+
+/**
+ * Asks the arduino to enable/disable a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ * @param {boolean} [enabled]
+ */
+
+Board.prototype.accelStepperEnable = function(deviceNum, enabled) {
+
+  if (typeof enabled === "undefined") {
+    enabled = true;
+  }
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x04, // ENABLE from firmware
+    deviceNum,
+    enabled & 0x01,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Asks the arduino to stop a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ */
+
+Board.prototype.accelStepperStop = function(deviceNum, callback) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x05, // STEPPER_STOP from firmware
+    deviceNum,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+
+  if (callback) {
```

Comment:
Please remove this callback. I was wrong in requesting it be added to `accelStepperStop`. This results in the callback being called twice (once because it's registered when a step sequence starts and a second time when it stops).

Reply:
I can't tell if this was addressed?

Reply:
This was resolved. The request was to remove the callback from accelStepperStop (which has been removed). It looks like I had added this comment to the wrong method.

Reply:
Cool, thanks for clarifying